invenio_rdm_records/jsonschemas/records/record-v1.0.0.json

```diff
@@ -69,64 +299,90 @@
-              "additionalProperties": false,
-              "type": "object",
-              "properties": {
-                "source": {
-                  "type": "string"
-                },
-                "value": {
-                  "type": "string"
-                }
+          "type": "object",
+          "properties": {
+              "description": {
+                "description": "Description/abstract for record.",
+                "type": "string"
+              },
+              "type": {
+                "description": "Type of description.",
+                "type": "string"
+              },
+              "lang": {
+                "description": "Language of the description. ISO 639-3 language code.",
+                "type": "string",
+                "maxLength": 3
               }
-            }
           },
-          "name": {
-            "description": "Full name of person or organisation. Personal name format: family, given.",
-            "type": "string"
-          },
-          "affiliations": {
-            "description": "Affiliation(s) for the purpose of this specific record.",
-            "type": "array",
-            "uniqueItems": true,
-            "items": {
+          "required": ["description"]
+      }
+    },
+    "publication_date": {
+      "description": "Record publication date (IS8601-formatted).",
+      "type": "string",
+      "format": "date"
+    },
+    "licenses": {
+      "description": "Any license or copyright information for this resource.",
+      "type": "array",
+      "uniqueItems": true,
+      "minItems": 0,
+      "items": {
+        "type": "object",
+        "properties": {
+            "license": {
+              "description": "The license itself. Free text.",
+              "type": "string"
+            },
+            "uri": {
+              "description": "The URI of the license.",
+              "type": "string",
+              "format": "uri"
+            },
+            "identifier": {
+              "description": "A short, standardized version of the license name.",
+              "type": "string"
+            },
+            "scheme": {
+              "description": "The name of the scheme.",
               "type": "string"
             }
+        }
+      }
+    },
+    "subjects": {
+      "type": "array",
+      "uniqueItems": true,
+      "minItems": 0,
+      "items": {
+          "type": "object",
+          "properties": {
+              "subject": {"type": "string"},
+              "identifier": {"type": "string"},
+              "scheme": {"type": "string"}
           },
-          "email": {
-            "type": "string",
-            "description": "Contact email for the purpose of this specific record.",
-            "format": "email"
-          },
-          "role": {
-            "description": "",
-            "type": "string"
-          }
-        },
-        "required": [
-          "name"
-        ]
+          "required": ["subject"]
```

Comment:
This seems like it would add a lot of weight to a submission form. If I want to allow user generated subjects, I need code that's going to generate a unique identifier/compare against other user generated subjects? If that's in place, sure, but otherwise just requiring subject makes more sense.

Reply:
> compare against other user generated subjects

I think that is in fact the sole value proposition of user-generated keywords. Otherwise might as well put those keywords in the description: it would be the same and we would truly not need any additional code. We are going to need standardization/saving code to have subjects become reusable and facet appropriately, but it shouldn't be too much trouble. We developed something similar just for ingestion of regular subject terms.